bokehjs/src/coffee/models/annotations/tooltip.coffee

```diff
@@ -70,11 +75,11 @@ export class TooltipView extends AnnotationView
         @el.classList.add("bk-right")
         left = sx - @el.offsetWidth - arrow_size
         top = sy - @el.offsetHeight/2
-      when "above"
+      when "below"
         @el.classList.add("bk-above")
         top = sy + (@el.offsetHeight - @el.clientHeight) + arrow_size
         left = Math.round(sx - @el.offsetWidth/2)
-      when "below"
+      when "above"
```

Comment:
@mattpap this was the real problem. With this change, side is always relative to the point being annotated